homeassistant/components/climate/generic_thermostat.py

```diff
@@ -217,4 +217,4 @@ def _control_heating(self):
     @property
     def _is_device_active(self):
         """If the toggleable device is currently active."""
-        return switch.is_on(self.hass, self.heater_entity_id)
+        return switch.async_is_on(self.hass, self.heater_entity_id)
```

Comment:
You're right. I was only looking at `current_operation`, which is being called async (via `async_update_ha_state`). Will fix it.